Linux_cmake/nheqminer_cpu/CMakeLists.txt

```diff
@@ -0,0 +1,137 @@
+project(nheqminer_cpu)
+cmake_minimum_required(VERSION 2.8)
+
+#aux_source_directory(. SRC_LIST)
+#add_executable(${PROJECT_NAME} ${SRC_LIST})
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+
+# LINUX
+if(CMAKE_COMPILER_IS_GNUCXX)
+    # use native cpu features
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=native")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native")
+    # optimizations
+    add_definitions(-O3)
+endif()
+
+# Common
+include_directories(${nheqminer_SOURCE_DIR})
+
+add_definitions(-DBOOST_ALL_NO_LIB -DBOOST_ALL_DYN_LINK -DBOOST_LOG_DYN_LINK)
+
+find_package(Threads REQUIRED COMPONENTS)
+find_package(Boost REQUIRED COMPONENTS system log_setup log date_time filesystem thread)
+
+if (Boost_FOUND)
+  # From the offical documentation:
+  # Add include directories to the build. [...] If the SYSTEM option is given,
+  # the compiler will be told the directories are meant as system include
+  # directories on some platforms (signalling this setting might achieve effects
+  # such as the compiler skipping warnings [...])."
+  include_directories (SYSTEM ${Boost_INCLUDE_DIR})
+
+  # From the offical documentation:
+  # "Specify directories in which the linker will look for libraries. [...] Note
+  # that this command is rarely necessary. Library locations returned by
+  # find_package() and find_library() are absolute paths. Pass these absolute
+  # library file paths directly to the target_link_libraries() command. CMake
+  # will ensure the linker finds them."
+  link_directories (${Boost_LIBRARY_DIRS})
+else()
+    message("Boost_FOUND NOT FOUND")
+endif ()
+
+## Add solvers here
+add_definitions(-DUSE_CPU_XENONCAT)
+add_definitions(-DUSE_CPU_TROMP)
```

Comment:
the `TROMP` miner compiles fine here, but crashes during the run (test):

```
$ ./nheqminer_cpu_tromp -b

        ==================== www.nicehash.com ====================
                Equihash CPU&GPU Miner for NiceHash v0.3a
        Thanks to Zcash developers for providing base of the code.
            Special thanks to tromp, xenoncat and eXtremal-ik7 for providing
                 optimized CPU, CUDA and AMD equihash solvers .
        ==================== www.nicehash.com ====================

Setting log level to 2
[14:15:38][0x00007f47d9151740] Using SSE2: YES
[14:15:38][0x00007f47d9151740] Using AVX: YES
[14:15:38][0x00007f47d9151740] Using AVX2: NO
[14:15:38][0x00007f47d9151740] Benchmarking CPU worker (SolverStub)
[14:15:38][0x00007f47d9151740] Benchmarking CPU worker (SolverStub)
[14:15:38][0x00007f47d9151740] Benchmarking CPU worker (SolverStub)
[14:15:38][0x00007f47d9151740] Benchmarking CPU worker (SolverStub)
[14:15:38][0x00007f47d9151740] Benchmark starting... this may take several minutes, please wait...
[14:15:38][0x00007f47d9151740] Benchmark done!
[14:15:38][0x00007f47d9151740] Total time : 1 ms
[14:15:38][0x00007f47d9151740] Total iterations: 200
[14:15:38][0x00007f47d9151740] Total solutions found: 0
[14:15:38][0x00007f47d9151740] Speed: 200000 I/s
[14:15:38][0x00007f47d9151740] Speed: 0 Sols/s
```